tiledb/sm/c_api/tiledb.cc

```diff
@@ -1464,35 +1458,32 @@ capi_return_t tiledb_array_delete_fragments_v2(
     uint64_t timestamp_end) {
   auto uri = tiledb::sm::URI(uri_str);
   if (uri.is_invalid()) {
-    throw api::CAPIStatusException(
-        "Failed to delete fragments; Invalid input uri");
+    throw api::CAPIException("Failed to delete fragments; Invalid input uri");
   }
 
   // Allocate an array object
-  tiledb::sm::Array array(ctx->resources(), uri);
+  auto array = make_shared<tiledb::sm::Array>(HERE(), ctx->resources(), uri);
 
   // Set array open timestamps
-  array.set_timestamp_start(timestamp_start);
-  array.set_timestamp_end(timestamp_end);
+  array->set_timestamp_start(timestamp_start);
+  array->set_timestamp_end(timestamp_end);
 
   // Open the array for exclusive modification
-  throw_if_not_ok(array.open(
+  throw_if_not_ok(array->open(
       static_cast<tiledb::sm::QueryType>(TILEDB_MODIFY_EXCLUSIVE),
       static_cast<tiledb::sm::EncryptionType>(TILEDB_NO_ENCRYPTION),
       nullptr,
       0));
 
   // Delete fragments
   try {
-    array.delete_fragments(uri, timestamp_start, timestamp_end);
+    array->delete_fragments(uri, timestamp_start, timestamp_end);
   } catch (...) {
-    throw_if_not_ok(array.close());
-    std::throw_with_nested(
-        api::CAPIStatusException("Failed to delete fragments"));
+    throw_if_not_ok(array->close());
```

Comment:
Need `throw;` here, otherwise execution continues.

Reply:
Done. I was confused the first time by what you had meant.